examples/rtic.rs

```diff
@@ -0,0 +1,48 @@
+#![deny(warnings)]
+#![no_std]
+#![no_main]
+
+extern crate rtic;
+
+use panic_halt as _;
+use rtic::app;
+use stm32f4xx_hal::{
+    gpio::{gpioa::PA0, gpiod::PD12, Edge, Input, Output, PullDown, PushPull},
+    prelude::*,
+    stm32,
+};
+
+#[app(device = stm32f4xx_hal::stm32, peripherals = true)]
+const APP: () = {
+    struct Resources {
+        button: PA0<Input<PullDown>>,
+        led: PD12<Output<PushPull>>,
+    }
+
+    #[init]
+    fn init(mut ctx: init::Context) -> init::LateResources {
+        let mut syscfg = ctx.device.SYSCFG.constrain();
+
+        let gpiod = ctx.device.GPIOD.split();
+        let led = gpiod.pd12.into_push_pull_output();
+
+        let gpioa = ctx.device.GPIOA.split();
+        let mut button = gpioa.pa0.into_pull_down_input();
+        button.make_interrupt_source(&mut syscfg);
+        button.enable_interrupt(&mut ctx.device.EXTI);
+        button.trigger_on_edge(&mut ctx.device.EXTI, Edge::RISING);
+
+        stm32::NVIC::unpend(stm32::Interrupt::EXTI0);
+        unsafe {
+            stm32::NVIC::unmask(stm32::Interrupt::EXTI0);
+        };
```

Comment:
The RTIC-generated code unmasks the interrupt before calling init, so we don't need to do that again here.

Reply:
Oh, I missed that. Thanks for the review. Fixed